vm/vm.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package vm
+
+import (
+	"github.com/DE-labtory/koa/opcode"
+	"github.com/pkg/errors"
+)
+
+var opcodes = map[opcode.Type]Opcode{
+	opcode.Add:  Add{},
+	opcode.Push: Push{},
+}
+
+type Item uint32
+
```

Comment:
imo `Item` too comprehensive

Reply:
instruction has opcodes and datas. I think Item could represent both. But on second thoughts, uint32 will be better because the processing of opcodes and datas is separated. What do you think?

Reply:
What about `Instruction`? 
```go
func Execute(instructions []Item) (Stack, error) {
```

```go
type InstructionBuffer interface {
 	GetItems(size int) []Item
 }
```
I think in both case `Instruction` might be better than `Item` imo 👍